### DISCUSSION TITLE
Vercel Authentication within an iframe

### DISCUSSION THREAD

**Discussion by jinwookie** [vercel/community#1664] [Help] [answered]
### Summary

We are testing a site within an iframe, that is currently behind Vercel Authentication. However, when hitting the authentication, we are getting an error, 

`Refused to display 'https://site' in a frame because it set 'X-Frame-Options' to 'deny'.`

I have tried a few bypass options, using the shareable which is in beta, and the protection bypass by adding the querystring params, but those still return the same error.

I turned off the authentication, and the site can load within the iframe fine.

**Comment by amyegan** [comment]
Hi Brian. The `X-Frame-Options:DENY` header is on the password page to avoid [click-jacking](https://developer.mozilla.org/en-US/docs/Web/Security/Types_of_attacks#click-jacking) attacks. There is currently not a way to change this header, but I'm sharing your situation with the engineering team

**Comment by roch-numbered** [comment]
Hi,
I faced the same problem and while digging through the documentation of [Password Protection](https://vercel.com/docs/concepts/deployments/advanced-deployment-protection#password-protection) I found out you can [bypass it](https://vercel.com/docs/concepts/deployments/advanced-deployment-protection#protection-bypass-for-automation).
It offers two headers, `x-vercel-protection-bypass` & `x-vercel-set-bypass-cookie`, that are also available as query parameters.

However, using only `?x-vercel-protection-bypass=<secret>` doesn't load CSS/JS/images inside the iframe and combining it with `?x-vercel-protection-bypass=<secret>&x-vercel-set-bypass-cookie=true` doesn't display the website anymore (showing back the deny message).

Those headers/parameters seem to be what we are looking for, though they aren't working correctly when used as query parameters.

I'm patiently awaiting updates and fixes on this, thx 🤓

**Comment by amyegan** [comment, marked as answer]
Thanks again for bringing this up @jinwookie! You can now use `x-vercel-set-bypass-cookie: samesitenone` to allow indirect access to your deployment via iframe.  More info is available in the [Protection Bypass for Automation](https://vercel.com/docs/concepts/deployments/advanced-deployment-protection#using-protection-bypass-for-automation) documentation
